src/Storage/Storage/Common/AzureStorageTable.cs

```diff
@@ -37,14 +42,67 @@ public class AzureStorageTable : AzureStorageBase
         public Uri Uri { get; private set; }
 
         /// <summary>
-        /// Azure storage table constructor
+        /// Constructs AzureStorageTable from track 1 CloudTable.
+        /// Internally it constructs track 2 TableClient used by data plane cmdlets.
+        /// </summary>
+        /// <param name="table">Cloud table object.</param>
+        /// <param name="storageContext">Storage context containing account information used to construct TableClient.</param>
+        /// <param name="options">Table client options which should contain powershell user agent.</param>
+        public AzureStorageTable(CloudTable table, AzureStorageContext storageContext, TableClientOptions options)
+        {
+            this.CloudTable = table;
+            this.Name = table.Name;
+            this.Uri = table.Uri;
+            
+            this.Context = storageContext;
+
+            // construct track 2 table client from track 1 cloud table instance
+            this.TableClient = AzureStorageTable.ConstructTableClientFromCloudTable(this.CloudTable, options);
+        }
+
+        /// <summary>
+        /// Constructs AzureSTorageTable from track 2 TableClient.
+        /// Internally it construcs track 1 CloudTable for property display purpose only.
         /// </summary>
-        /// <param name="table">Cloud table object</param>
-        public AzureStorageTable(CloudTable table)
+        /// <param name="tableClient"></param>
+        public AzureStorageTable(TableClient tableClient)
+        {
+            this.TableClient = TableClient;
+            this.Name = tableClient.Name;
+
+            Uri uri = (Uri)typeof(TableClient)
+                .GetField("_endpoint", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance)
+                .GetValue(tableClient);
+            this.Uri = new Uri($"{uri.AbsoluteUri}{(uri.AbsoluteUri.EndsWith("/") ? string.Empty : "/")}{Name}");
+
+            // This constructed CloudTable is only for display purpose.
+            // Without credential content, it wouldn't work for data plane oprations.
+            // Customer, who uses oauth, should use track 2 TableClient instead.
+            this.CloudTable = new CloudTable(Uri);
```

Comment:
Have you tested if use oauth run Get-AzStorageTable, and use the output to run cmdlets in AzTable, what's the error message?
Will the error message be clear enough to let user know the root cause of the failure?

Reply:
Per my test, the error will be following, like not so clear to let customer know the root cause.
```
PS C:\temp>  Add-AzTableRow -table $storageTable -partitionKey $partitionKey1 -rowKey ("CA") -property @{"username"="Chris";"userid"=1;"DateTimeProperty"="2012-01-02T23:00:00";"DateTimeProperty@odata.type"="Edm.DateTime"}
Exception calling "Execute" with "1" argument(s): "The specified resource does not exist."
At C:\Program Files\WindowsPowerShell\Modules\AzTable\2.1.0\AzureRmStorageTableCoreHelper.psm1:266 char:10
+ ...      return ($Table.Execute([Microsoft.Azure.Cosmos.Table.TableOperat ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : StorageException
```

Reply:
AzTable doesn't use this AzureStorageTable.cs
https://github.com/paulomarquesc/AzureRmStorageTable/blob/7d6b260d89c42d0496a53a6b9f300cfef01e7e3d/AzureRmStorageTableCoreHelper.psm1#L182

Reply:
That'll be the error msg as using storage table based off of oauth token in data plane operations are not supported.
To customize error message needs to be implemented in the data plane az-table module, which is not planned.
The next step is to implement data plane cmdlets here and deprecate az-table.